discord/abc.py

```diff
@@ -280,6 +281,11 @@ async def _edit(self, options, reason):
                 perms.append(payload)
             options['permission_overwrites'] = perms
 
+        ch_type = options.get('type', self.type)
+        if not isinstance(ch_type, ChannelType):
+            raise InvalidArgument('type field must be of type ChannelType')
+        options['type'] = ch_type.value
```

Comment:
I think this implementation is too eager (it sends the `type` key all the time even if it doesn't actually change). Alternatively this implementation is better:

```suggestion
        try:
            ch_type = options['type']
        except KeyError:
            pass
        else:
            if not isinstance(ch_type, ChannelType):
                raise InvalidArgument('type field must be of type ChannelType')
            options['type'] = ch_type.value
```